src/app/features/auth/login/login.component.ts

```diff
@@ -0,0 +1,88 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { AbstractControl, FormGroup, NonNullableFormBuilder, ReactiveFormsModule, Validators } from "@angular/forms";
+import { Router } from "@angular/router";
+import { AuthService } from "../services/auth.service";
+import { StorageService } from "../../../core/services/storage.service";
+import { Store } from "@ngxs/store";
+import { LoginAction, LoginTokenAction } from "@src/app/state/auth.action";
+import { Token } from "@src/app/shared/models/token";
+import { take } from "rxjs";
+
+@Component({
+  selector: 'login',
+  standalone: true,
+  imports: [CommonModule, ReactiveFormsModule],
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LoginComponent implements OnInit {
+
+  public form: FormGroup;
+  public isSubmitted = false;
+
+  constructor(
+    private readonly fb: NonNullableFormBuilder,
+    private readonly router: Router,
+    private readonly authService: AuthService,
```

Comment:
If you follow [the architecture we discussed a lot](https://georgebyte.com/scalable-angular-app-architecture/), you'll find out that components should know nothing about HTTP services. Components only know about Store.